examples/demo/metro.config.js

```diff
@@ -1,12 +1,32 @@
-const {getDefaultConfig, mergeConfig} = require('@react-native/metro-config');
+const { getDefaultConfig } = require('@react-native/metro-config');
 
 const path = require('path');
 
+const projectRoot = __dirname;
+
+// If you have a monorepo, the workspace root may be above the project root.
+const workspaceRoot = path.resolve(projectRoot, "../..");
+const appAuthPackageRoot = path.resolve(workspaceRoot, "packages", "react-native-app-auth");
+
 const packagePath = path.resolve(
   path.join(process.cwd(), '..', '..', 'packages', 'react-native-app-auth'),
 );
 
+/**
+ * A crude flag to force Metro to resolve one or another fork of react-native.
+ * Set to "macos" to resolve react-native-macos.
+ * Set to "ios" to resolve react-native.
+ * Welcoming a fully automatic solution by anyone more familiar with Metro!
+ * @type {"macos" | "ios"}
+ */
+const applePlatform = "ios";
```

Comment:
This is the best I could do with the time I have available. If anyone's more familiar with Metro, please show me how we can get Metro to resolve the correct one out of `react-native-macos` and `react-native` in this yarn monorepo.

Reply:
Metro has some support for out of tree platforms, and https://github.com/microsoft/rnx-kit/tree/main/packages/metro-config has the correct defaults I think. It's what I have used for everywhere I added macOS support